geyser-plugin-interface/src/geyser_plugin_interface.rs

```diff
@@ -62,12 +62,32 @@ pub struct ReplicaTransactionInfo<'a> {
     pub transaction_status_meta: &'a TransactionStatusMeta,
 }
 
+/// Information about a transaction, including index in block
+#[derive(Clone, Debug)]
+pub struct ReplicaTransactionInfoWithIndex<'a> {
```

Comment:
I would just name this to ReplicaTransactionInfoV2 so that if we add more versions to it in the future, it will be easier to keep the names uniform. Otherwise, we would have to name it to like ReplicaTransactionInfoWithIndexAndWithX...